ambry-router/src/main/java/com/github/ambry/router/SimpleOperationTracker.java

```diff
@@ -176,6 +179,15 @@
         eligibleReplicas = getEligibleReplicas(partitionId, null,
             EnumSet.of(ReplicaState.BOOTSTRAP, ReplicaState.STANDBY, ReplicaState.LEADER));
         break;
+      case UndeleteOperation:
+        diskParallelism = routerConfig.routerUndeleteRequestParallelism;
+        crossColoEnabled = true;
+        eligibleReplicas = getEligibleReplicas(partitionId, null,
+            EnumSet.of(ReplicaState.BOOTSTRAP, ReplicaState.STANDBY, ReplicaState.LEADER));
```

Comment:
if there are machines that are offline, they won't be included in this set. Is it okay to ignore them? I believe this policy currently says that an undelete needs 100% success from replicas that are currently up.

Reply:
non-eligible hosts would be considered as a failure. for instance, if a dc has three hosts, and one is offline, then starting from the beginning, we already have a failure hosts. If we see another hosts returning a failure response, then the operation is considered as failed. But if the other two both return success, then undelete operation still is valid.

Reply:
Ah, I think I got confused by the setting of disk success target, but that seems to just be a dummy value and isn't used anywhere in UndeleteOperationTracker.